libvast/src/concept/parseable/vast/legacy_type.cpp

```diff
@@ -14,13 +14,20 @@
 #include "vast/concept/parseable/numeric/integral.hpp"
 #include "vast/concept/parseable/string/quoted_string.hpp"
 #include "vast/concept/parseable/vast/identifier.hpp"
+#include "vast/detail/string_literal.hpp"
 
 #include <caf/optional.hpp>
 
 namespace vast {
 
-template <class T>
+template <class T, detail::string_literal DeprecationNotice = "">
```

Comment:
This looks like a `detail::`'ish thing potentially useful in other places as well.

Reply:
Yes, but not for this PR.